lib/WebService/SendBird.pm

```diff
@@ -0,0 +1,306 @@
+
+More information at L<Platform API Documentation|https://docs.sendbird.com/platform>
+
+=cut
+
+use constant DEFAULT_API_URL_TEMPLATE => 'https://api-%s.sendbird.com/v3';
+use constant DEFAULT_REQUEST_TIMEOUT  => 15;
+
+=head1 METHODS
+
+=head2 new
+
+Creates an instance of API client
+
+=over 4
+
+=item * C<api_token> - Master or Secondary API Token.
+
+=item * C<app_id> - Application ID.
+
+=item * C<api_url> - URL to API end point. By default it will be generated from app_id.
+
+=item * C<ua> - Custom http client for API requests, should have the same interface like L<Mojo::UserAgent>.
+
+=item * C<timeout> - request timeout, default value 15 seconds
+
+=back
+
+=cut
+
+
+sub new {
+    my ($cls, %params) = @_;
+
+    Carp::croak('Missing required argument: api_token') unless $params{api_token};
+    Carp::croak('Missing required argument: app_id or api_url') unless $params{app_id} || $params{api_url};
+
+    my $self = +{
+        api_token => $params{api_token},
+        $params{ua}      ? (ua      => $params{ua})                      : (),
+        $params{app_id}  ? (app_id  => $params{app_id})                  : (),
+        $params{api_url} ? (api_url => Mojo::URL->new($params{api_url})) : (),
+    };
+
+    return bless $self, $cls;
+}
+
+
+=head2 app_id
+
+Returns Application ID.
+
+=cut
+
+sub app_id { shift->{app_id} }
+
+=head2 api_token
+
+Returns API Token
+
+=cut
+
+sub api_token { shift->{api_token} }
+
+=head2 api_url
+
+Returns API endpoint url
+
+=cut
+
+sub api_url {
+    my $self = shift;
+
+    $self->{api_url} //= Mojo::URL->new(sprintf(DEFAULT_API_URL_TEMPLATE, $self->app_id));
+
+    return $self->{api_url};
+}
+
+=head2 timeout
+
+Return http request timeout value.
+
+=cut
+
+sub timeout { shift->{timeout} }
+
+=head2 ua
+
+Return User Agent for http request.
+
+=cut
+
+sub ua {
+    my $self = shift;
+
+    return $self->{ua} if $self->{ua};
+
+    my $ua = Mojo::UserAgent->new();
+
+    $ua->inactivity_timeout($self->timeout || DEFAULT_REQUEST_TIMEOUT);
+    $ua->proxy->detect;
+    $ua->max_connections(100);
+
+    $self->{ua} = $ua;
+    return $self->{ua};
+}
+
+=head2 http_headers
+
+Returns headers for API request.
+
+=cut
+
+sub http_headers {
+    my $self = shift;
+
+    return {
+        'Content-Type' => 'application/json, charset=utf8',
+        'Api-Token'    => $self->api_token,
+    }
+}
+
+
+=head2 request
+
+Sends request to Sendbird API
+
+=cut
+
+sub request {
+    my ($self, $method, $path, $params) = @_;
+
+    my $resp = $self->ua->start(
+        $self->ua->build_tx(
+            $method,
+            $self->_url_for($path),
+            $self->http_headers,
+            uc($method) eq 'GET' ? (form => $params) : (json => $params),
+        )
+    );
+    #TODO Improve error handling
+    Carp::croak('Fail to make request to SB API') if $resp->result->code !~ /^2\d+/;
+
+    my $data = $resp->result->json;
```

Comment:
Suggest to wrap this in try... catch in case we get invalid json.

Reply:
Fixed.